npm/webpack-dev-server/test-wds-3.js

```diff
@@ -0,0 +1,49 @@
+const execa = require('execa')
+const pkg = require('./package.json')
+const fs = require('fs')
+
+/**
+ * This file installs WebpackDevServer 3 and runs the tests for the dev-server.
+ * We read package.json, update the webpack version, then re-run yarn install.
+ * After it finishes, pass or fail,
+ * we revert the package.json back to the original state.
+ *
+ * The tests for the example projects (inside of examples) run with WebpackDevServer 3.
+ * This ensures we have some coverage for both versions.
+ */
+const main = async () => {
+  const originalPkg = JSON.stringify(pkg, null, 2)
+
+  const resetPkg = async () => {
+    fs.writeFileSync('package.json', originalPkg, 'utf8')
+    await execa('yarn', ['install'], { stdio: 'inherit' })
+  }
+
+  const checkExit = async ({ exitCode }) => {
+    if (typeof exitCode !== 'number') {
+      // eslint-disable-next-line no-console
+      console.error(`Finished with missing exit code from execa (received ${exitCode})`)
+    }
+
+    await resetPkg()
+    process.exit(exitCode)
+  }
+
+  pkg.devDependencies['webpack-dev-server'] = '^3.11.0'
```

Comment:
Hm, should we specify the exact version during CI runs to ensure the tests always us the same one? eg `3.11.0` (no `^`)

Reply:
Makes sense I'll pin it.